code/controllers/subsystem/machines.dm

```diff
@@ -79,19 +82,49 @@ SUBSYSTEM_DEF(machines)
 	if (!resumed)
 		for(var/datum/powernet/powernet as anything in powernets)
 			powernet.reset() //reset the power state.
-		src.currentrun = processing.Copy()
+		current_part = SSMACHINES_APCS_EARLY
+		src.currentrun = apc_early_processing.Copy()
 
 	//cache for sanic speed (lists are references anyways)
 	var/list/currentrun = src.currentrun
 
-	while(currentrun.len)
-		var/obj/machinery/thing = currentrun[currentrun.len]
-		currentrun.len--
-		if(QDELETED(thing) || thing.process(wait * 0.1) == PROCESS_KILL)
-			processing -= thing
-			thing.datum_flags &= ~DF_ISPROCESSING
-		if (MC_TICK_CHECK)
-			return
+	//APC early processing. Draws static power usages from their grids.
+	if(current_part == SSMACHINES_APCS_EARLY)
+		while(currentrun.len)
+			var/obj/machinery/power/apc/apc = currentrun[currentrun.len]
+			currentrun.len--
+			if(QDELETED(apc) || apc.early_process(wait * 0.1) == PROCESS_KILL)
+				apc_early_processing -= apc
+				apc.datum_flags &= ~DF_ISPROCESSING
+			if(MC_TICK_CHECK)
+				return
+		current_part = SSMACHINES_MACHINES
+		currentrun = processing.Copy()
+
+	//General machine processing. Their power usage can be dynamic and based on surplus power, so they come after static power usage have been applied.
+	if(current_part == SSMACHINES_MACHINES)
+		while(currentrun.len)
+			var/obj/machinery/thing = currentrun[currentrun.len]
+			currentrun.len--
+			if(QDELETED(thing) || thing.process(wait * 0.1) == PROCESS_KILL)
+				processing -= thing
+				thing.datum_flags &= ~DF_ISPROCESSING
+			if (MC_TICK_CHECK)
+				return
+		current_part = SSMACHINES_APCS_LATE
+		currentrun = apc_late_processing.Copy()
+
+	//APC late processing. APCs will use the remaining power on the grid to charge their cells if needed.
+	//This is applied at the end so charging APCs don't cause others to discharge by taking all the power from the grid before machines use power.
+	if(current_part == SSMACHINES_APCS_LATE)
+		while(currentrun.len)
+			var/obj/machinery/power/apc/apc = currentrun[currentrun.len]
+			currentrun.len--
+			if(QDELETED(apc) || apc.late_process(wait * 0.1) == PROCESS_KILL)
+				apc_late_processing -= apc
+				apc.datum_flags &= ~DF_ISPROCESSING
+			if(MC_TICK_CHECK)
+				return
```

Comment:
I noticed while looking at the vareditor that the APCs' cells are being used by small amounts while others are charging (enough power to supply machines) when load gets maxed by APC charge. Why is this the case?